### PR TITLE
hist/plt updates + bug fixes

### PR DESCRIPTION
**Pull request by pmlugato**
worked with Joey and Luca on the following changes:
- added missing xsec for inclusive TTJets sample
- added single filelist with bkgd and signal for use at plotting stage (need to submit private signals and bkgds separately in producer)
- when making hists, new variables added THEN selections made (so you can select on new variables)
- added hists relevant to WH in hist_defs
- pileup, prefire, and PS weights applied for WH (previous revamp of plotting flow reverted this)
- TopPT -> HighestPT in make_hists config files
- xsecs applied correctly for WH private samples as well now
- saving git diffs works for submit.py now
- minor bug fixes/old comment removal

### CODE REVIEW

**Review by lucalavezzo — APPROVED** [review]

File: histmaker/submit.py [hunk 1]
```diff
@@ -109,9 +109,9 @@ def call_process(cmd, work_dir):
         getpass.getuser(), np.random.randint(0, 10000)
     )
     os.system(f"mkdir {work_dir}")
-    os.system(f"cp -R ../* {work_dir}/.")
+    os.system(f"cp -a ../../SUEPCoffea_dask {work_dir}/.")
```
Comment (chadfreer) [inline histmaker/submit.py:112]: was there a problem with permissions to prompt this change. No fundamental issue but I am curious.

File: plotting/plot_utils.py [hunk 2]
```diff
@@ -363,8 +363,8 @@ def loader(
         # get the normalization factor for SUEP samples
         if xsec_SUEP:
             sample_name = infile_name.split("/")[-1].split("13TeV")[0] + "13TeV-pythia8"
-            if (
-                "SUEP" in sample_name
+            if ("SUEP" in sample_name) or (
```
Comment (chadfreer) [inline plotting/plot_utils.py:366]: see comment above

File: histmaker/submit.py (hunk 1 above)
Comment (lucalavezzo) [reply histmaker/submit.py:112]: yea, the hidden files were not copying over, which breaks when we try to save the git diff.

File: histmaker/make_hists.py [hunk 3]
```diff
@@ -583,7 +590,7 @@ def main():
     if options.isMC:
         logging.info("Applying normalization.")
         xsection = 1
-        if "SUEP" not in sample:
+        if ("SUEP" not in sample) or ("WHlepton" not in sample):
```
[the remark this answers is not in the document]
Comment (pmlugato) [reply histmaker/make_hists.py:593]: are these the central samples? the private samples we have been using don't have "SUEP" in the name, but "WHlepton"

File: data/xsections_2018.json [hunk 4]
```diff
@@ -269,6 +269,11 @@
     "br": 1,
     "kr": 1
   },
+  "TTJets_TuneCP5_13TeV-amcatnloFXFX-pythia8+RunIISummer20UL18MiniAODv2-106X_upgrade2018_realistic_v16_L1v1-v2+MINIAODSIM": {
+    "xsec": 831.76,
+    "br": 1,
+    "kr": 1
+  },
```
Comment (jreic) [inline data/xsections_2018.json:276]: I suggest that we add it also for 16+17, so we don't forget later on (true for any other samples too, but we can start with this).

File: histmaker/make_hists.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jreic) [reply histmaker/make_hists.py]: I could mass rename the private samples to add SUEP in the name, though it wouldn't help for backwards compatibility (and we won't be relying on them as much going forward anyway, I hope)

File: data/xsections_2018.json (hunk 4 above)
Comment (pmlugato) [reply data/xsections_2018.json:276]: We discovered that we didn't produce inclusive TTJets samples for 2016 and 2017, so we will need to produce these (and then we can add them into the xsec jsons)

Comment (jreic) [reply data/xsections_2018.json:276]: Sounds good, we can deal with it later then!

**Review by jreic — APPROVED** [review]

**Pull request merged**